src/styles/calendar.scss

```diff
@@ -0,0 +1,87 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: 1px;
+  width: calc(#{$days-in-week} * #{$cell-size}
+  + (#{$days-in-week} - 1) * #{$border-width} + 2 * #{$cell-padding});
+
+  &__day {
+    width: #{$cell-size};
+    height: #{$cell-size};
+    border: #{$border-width} solid black;
+    background-color: #{$color-white};
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    position: relative;
+    cursor: pointer;
+    font-size: #{$cell-font-size};
+    font-family: #{$num-font-family};
+    transition: transform #{$animation-duration} ease-in-out;
+
+    @for $dayNumber from 1 through 31 {
+      &:nth-child(#{$dayNumber})::before {
+        content: "#{$dayNumber}";
+      }
+    }
+  }
+
+  &__day:hover {
+    background-color: #{$color-pink};
+    transform: translateY(-20px);
+  }
+}
+
+@each $day in mon,
+  tue,
+  wed,
+  thu,
+  fri,
+  sat,
+  sun {
+  .start-day--#{$day} {
+    .calendar__day {
+      &:nth-child(#{$days-in-week}n + 1) {
+        margin-left: 0;
+      }
+
+      &:nth-child(#{$days-in-week}n + 2) {
+        margin-left: calc(#{$cell-size} +
+        #{$border-width} + 2 * #{$cell-padding});
+      }
+
+      &:nth-child(#{$days-in-week}n + 3) {
+        margin-left: calc(2 * (#{$cell-size} +
+        #{$border-width} + 2 * #{$cell-padding}));
+      }
+
+      &:nth-child(#{$days-in-week}n + 4) {
+        margin-left: calc(3 * (#{$cell-size} +
+        #{$border-width} + 2 * #{$cell-padding}));
+      }
+
+      &:nth-child(#{$days-in-week}n + 5) {
+        margin-left: calc(4 * (#{$cell-size} +
+        #{$border-width} + 2 * #{$cell-padding}));
+      }
+
+      &:nth-child(#{$days-in-week}n + 6) {
+        margin-left: calc(5 * (#{$cell-size} +
+        #{$border-width} + 2 * #{$cell-padding}));
+      }
+
+      &:nth-child(#{$days-in-week}n + 7) {
+        margin-left: calc(6 * (#{$cell-size} +
+        #{$border-width} + 2 * #{$cell-padding}));
+      }
+    }
+  }
+}
```

Comment:
Really good job, but you can simplify it. 
```
$days: (
  "mon": 1,
  "tue": 2,
  "wed": 3,
  "thu": 4,
  "fri": 5,
  "sat": 6,
  "sun": 7
);
```
Start with it and try to use the loop for the other.